todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/local/TasksLocalDataSource.kt

```diff
@@ -159,25 +102,29 @@ class TasksLocalDataSource private constructor(context: Context) : TasksDataSour
     }
 
     override fun deleteAllTasks() {
-        with(dbHelper.writableDatabase) {
-            delete(TABLE_NAME, null, null)
-            close()
-        }
+        appExecutors.diskIO.execute { tasksDao.deleteTasks() }
     }
 
     override fun deleteTask(taskId: String) {
-        val selection = "$COLUMN_NAME_ENTRY_ID LIKE ?"
-        val selectionArgs = arrayOf(taskId)
-        with(dbHelper.writableDatabase) {
-            delete(TABLE_NAME, selection, selectionArgs)
-            close()
-        }
+        appExecutors.diskIO.execute { tasksDao.deleteTaskById(taskId) }
     }
 
     companion object {
         private var INSTANCE: TasksLocalDataSource? = null
 
-        @JvmStatic fun getInstance(context: Context) =
-                INSTANCE ?: TasksLocalDataSource(context).apply { INSTANCE = this }
+        @JvmStatic
+        fun getInstance(appExecutors: AppExecutors, tasksDao: TasksDao): TasksLocalDataSource {
+            if (INSTANCE == null) {
+                synchronized(TasksLocalDataSource::javaClass) {
+                    INSTANCE = TasksLocalDataSource(appExecutors, tasksDao)
+                }
+            }
+            return INSTANCE!!
+        }
+
+        @VisibleForTesting
+        fun clearInstance() {
```

Comment:
expression

Reply:
Method with body `INSTANCE = null` can't be made into an expression.

Reply:
My bad, please ignore.